### ISSUE TITLE
Improve API documentation for twisted/lore/tree.py

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2301]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2301|
|Type|enhancement|
|Created|2006-12-17 23:10:06Z|

tree.py implements the bulk of lore's functionality.  Much of it is quite opaque, however, due to inconsistent style and a near complete lack of maintainer documentation.

Adding docstrings to all of the potentially public API exposed by this module is a first step towards a reasonable maintainable lore.

<details><summary>Searchable metadata</summary>

```
trac-id__2301 2301
type__enhancement enhancement
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__lore lore
keywords__ 
time__1166397006000000 1166397006000000
changetime__1166579205000000 1166579205000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun removed owner|
|-|-|

Ready for review in lore-tree-docs-2301

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph set owner to @exarkun|
|-|-|

There is absolutely no possible basis for rejecting this branch, even if the new docstrings were nothing but profanity they would probably be better than what's there now.  Please merge this immediately!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set status to `closed`|
|-|-|

(In [18953]) Merge lore-tree-docs-2301

Author: exarkun
Reviewer: glyph
Fixes [#2301](https://github.com/twisted/twisted/issues/2301)

Add docstrings to all of the potentially-public functions defined in
twisted/lore/tree.py.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
